python/task_manager/worker_thread.py

```diff
@@ -52,8 +52,8 @@ def shut_down(self):
         """
         Shut down the thread and wait for it to exit before returning
         """
-        self._results_dispatcher = None
         with self._mutex:
+            self._results_dispatcher = None
```

Comment:
This actually fixed a small race condition when tests finalized. Sometimes the dispatcher would become None in other parts of the code while the mutex was taken, which is a no-no, considering it's role is to provide thread-safe access to it!